library/std/src/sys/unix/mod.rs

```diff
@@ -151,9 +152,25 @@ pub unsafe fn init(argc: isize, argv: *const *const u8) {
         }
     }
 
-    unsafe fn reset_sigpipe() {
+    unsafe fn reset_sigpipe(#[allow(unused_variables)] sigpipe: u8) {
         #[cfg(not(any(target_os = "emscripten", target_os = "fuchsia", target_os = "horizon")))]
-        rtassert!(signal(libc::SIGPIPE, libc::SIG_IGN) != libc::SIG_ERR);
+        {
+            // We don't want to add this as a public type to libstd, nor do we want to
+            // duplicate the code, so we choose to include this compiler file like this.
+            mod sigpipe {
+                include!("../../../../../compiler/rustc_session/src/config/sigpipe.rs");
+            }
```

Comment:
Uh no please don't. This will break building the standard library from the rust-src component.

Reply:
(which will break Miri and xargo, and maybe -Zbuild-std -- not sure about the latter)

Reply:
Thank you for pointing that out! It would be good to add proper checks to CI to detect such problems, like building Miri, or similar. Probably not an original idea though...

I agree with Josh that simply duplicating these constants with cross-references is the best solution (we only need to duplicate 3 of them actually). I kept `mod sigpipe` so that if you `git grep sigpipe::SIG_IGN` you get a hit both in std and in compiler. And it becomes more of a "literal" inlining of the constants, which I like.

Reply:
Miri is built by bors CI. However Miri is allowed to break since it is an externally tracked tool, and it is red right now, meaning CI would not have noticed a status change.